src/lib/y2partitioner/widgets/lvm_vg_devices_selector.rb

```diff
@@ -92,53 +91,76 @@ def unselect(sids)
-      # @macro seeAbstractWidget
-      #
-      # @see #selected_devices_validation
-      # @see #size_validation
       #
-      # @note An error popup is shown when there is some error in selected devices.
+      # An error popup is shown when there is some error.
       #
       # @return [Boolean]
       def validate
-        selected_devices_validation && size_validation
+        errors = self.errors
+
+        return true if errors.none?
+
+        Yast2::Popup.show(errors.first, headline: :error)
+        false
       end
 
       private
 
       # @return [Actions::Controllers::LvmVg]
       attr_reader :controller
 
-      # Validates that at least one physical volume was added to the volume group
+      # Errors when selecting physical volumes
       #
-      # @note An error popup is shown when no physical volume was added.
+      # @return [Array<String>]
+      def errors
+        [devices_error, size_error, striped_lvs_devices_error, striped_lvs_size_error].compact
+      end
+
+      # Error when no physical volume is selected
       #
-      # @return [Boolean]
-      def selected_devices_validation
-        return true if controller.devices_in_vg.size > 0
+      # @return [String, nil] nil if physical volumes are selected
+      def devices_error
+        return nil if controller.devices_in_vg.size > 0
 
-        # TRANSLATORS: Error message when no device is selected
-        Yast::Popup.Error(_("Select at least one device."))
-        false
+        # TRANSLATORS: Error message
+        _("Select at least one device.")
       end
 
-      # Validates that the volume group size is enough for all its logical volumes
+      # Error when the volume group size is not big enough to allocate all its logical volumes
       #
-      # @note An error popup is shown when the size is not enough.
+      # @return [String, nil] nil if the volume group is big enough
+      def size_error
+        return nil if controller.vg_size >= controller.lvs_size
+
+        # TRANSLATORS: Error message where %s is replaced by a size (e.g., 4 GiB).
+        format(_("The volume group size cannot be less than %s."), controller.lvs_size)
+      end
+
+      # Error message when the selected physical volumes cannot allocate the striped volumes
       #
-      # @return [Boolean]
-      def size_validation
-        return true if controller.vg_size >= controller.lvs_size
-
-        Yast::Popup.Error(
-          # TRANSLATORS: Error message when the resulting volume group size is not
-          # enough. %s is replaced by a size (e.g., 4 GiB).
-          format(
-            _("The volume group size cannot be less than %s."), controller.lvs_size
-          )
+      # @return [String, nil]
+      def striped_lvs_devices_error
+        return nil if controller.devices_in_vg.size >= controller.max_stripes
+
+        format(
+          # TRANSLATORS: Error message, where %{max_stripes} is replaced by a number (e.g., 3).
```

Comment:
NP: when there is only one placeholder I would tend to use the syntax with `%s`, which is more familiar to translators.